Site_Files/src/main/java/com/google/sps/servlets/ChangeRating.java

```diff
@@ -0,0 +1,60 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.PostInformation;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.PrintWriter;
+
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+
+/** Servlet connect to the data store, store the comments data*/
+@WebServlet("/change-rating")
+public class ChangeRating extends HttpServlet {
+
+    @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.getWriter().println(request); 
+    response.getWriter().println(request.getParameter("id")); 
+    response.getWriter().println(request.getParameter("rating")); 
+    
+  }
+
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the server.
+    long id =  Long.parseLong(getParameter(request, "id", ""));
```

Comment:
What happens if one of the parameters isn't in the request? Technically this is a client error, but if we end up throwing an exception of some sort, that could trigger the HTTP 500 response you got. Worth double checking.

Reply:
Ah, I see you wrote your own method to prevent returning null if the value isn't set. That's good! In production code, we would probably go one step further and deliberately return an HTTP 4xx error of some sort, with a message yelling at the client that they forgot a parameter.